ext/nnstreamer/tensor_filter/tensor_filter_onnxruntime.cc

```diff
@@ -0,0 +1,564 @@
+  input_types.resize (inputInfo.num_tensors);
+
+  for (i = 0; i < inputInfo.num_tensors; i++) {
+
+    // Get input names
+    auto input_name = session.GetInputNameAllocated (i, allocator);
+    input_names_allocated_strings.push_back (std::move (input_name));
+    input_names.push_back (input_names_allocated_strings.back ().get ());
+
+    // Get input types
+    ONNXTensorElementDataType type
+        = session.GetInputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetElementType ();
+
+    if (getTensorType (type, &inputInfo.info[i].type)) {
+      cleanup ();
+      throw std::invalid_argument ("Failed to convert ONNXTensorElement intput data type");
+    }
+    input_types[i] = type;
+
+    // Get input shapes/dims
+    size_t num_dims
+        = session.GetInputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetDimensionsCount ();
+    input_shapes[i]
+        = session.GetInputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetShape ();
+
+    // free dimensions are treated as 1 if not overriden
+    for (int64_t &dim : input_shapes[i]) {
+      if (dim == -1) {
+        dim = 1;
+      }
+    }
+
+    if (getTensorDim (inputInfo.info[i].dimension, input_shapes[i], num_dims)) {
+      cleanup ();
+      throw std::invalid_argument ("Shape input rank too high.");
+    }
+
+    inputInfo.info[i].name = nullptr;
+
+    gchar *dim;
+    dim = gst_tensor_get_dimension_string (inputInfo.info[i].dimension);
+
+    nns_logd ("inputInfo[%zu] >> name[%s], type[%d], dim[%s]", i,
+        inputInfo.info[i].name, inputInfo.info[i].type, dim);
+    g_free (dim);
+  }
+
+  // initialize output info
+  output_shapes.resize (outputInfo.num_tensors);
+  output_types.resize (outputInfo.num_tensors);
+
+  for (i = 0; i < outputInfo.num_tensors; i++) {
+
+    // Get output node names
+    auto output_name = session.GetOutputNameAllocated (i, allocator);
+    output_names_allocated_strings.push_back (std::move (output_name));
+    output_names.push_back (output_names_allocated_strings.back ().get ());
+
+    // Get output node types
+    ONNXTensorElementDataType type
+        = session.GetOutputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetElementType ();
+
+    if (getTensorType (type, &outputInfo.info[i].type)) {
+      cleanup ();
+      throw std::invalid_argument ("Failed to convert ONNXTensorElement output data type");
+    }
+    output_types[i] = type;
+
+    // Get output shapes/dims
+    size_t num_dims
+        = session.GetOutputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetDimensionsCount ();
+    output_shapes[i]
+        = session.GetOutputTypeInfo (i).GetTensorTypeAndShapeInfo ().GetShape ();
+
+    // free dimensions are treated as 1 if not overriden
+    for (int64_t &dim : output_shapes[i]) {
+      if (dim == -1) {
+        dim = 1;
+      }
+    }
+
+    if (getTensorDim (outputInfo.info[i].dimension, output_shapes[i], num_dims)) {
+      cleanup ();
+      throw std::invalid_argument ("Shape output rank too high.");
```

Comment:
Q: Does it make an exception? If not, use ```std::runtime_error```.

Reply:
Thanks. updated it.